### ISSUE TITLE
Offset on SMPL 

### ISSUE THREAD

**Issue by MarilynKeller** [eth-ait/aitviewer#5]
With the current code, the generated AMASS sequence has feet below zero and an offset compared to the mocap markers :

![image](https://user-images.githubusercontent.com/102662583/181642611-c8a70537-c5b8-4d25-abb9-1ad03284b1f2.png)

This is caused by the change of coordinate to get Y axis up : https://github.com/eth-ait/aitviewer/blob/4aac6fd77bab30674db5858e83e0ae64f0a8a7d4/aitviewer/renderables/smpl.py#L177

I think the reason is that the rotation is applied to the root_joint of SMPL. Yet the root joint of SMPL is not the origin of SMPL (this detail is a common source of error). If you display SMPL in Tpose, the root joint is typically 20cm below the world origin. 

For now, I removed those lines in the `.from_amass` method and instead apply the rotation directly on the vertices in `fk()`

```
    def fk(self):
        """Get joints and/or vertices from the poses."""
        verts, joints = self.smpl_layer(poses_root=self.poses_root,
                                        poses_body=self.poses_body,
                                        poses_left_hand=self.poses_left_hand,
                                        poses_right_hand=self.poses_right_hand,
                                        betas=self.betas,
                                        trans=self.trans)
        skeleton = self.smpl_layer.skeletons()['body'].T
        faces = self.smpl_layer.bm.faces.astype(np.int64)
        joints = joints[:, :skeleton.shape[0]]

        #Transform output vertices and joints into our viewer's coordinate system (where Y is up).
        to_y_up = torch.Tensor([[1, 0, 0], [0, 0, 1], [0, -1, 0]]).to(self.device)
        verts = torch.matmul(to_y_up.unsqueeze(0), verts.unsqueeze(-1)).squeeze()
        joints = torch.matmul(to_y_up.unsqueeze(0), joints.unsqueeze(-1)).squeeze()
```

This fixes the offset issue:

![image](https://user-images.githubusercontent.com/102662583/181645276-990046ed-6741-4b31-a549-6598dd3315e3.png)

**Comment by kaufManu** [comment]
Thanks for reporting this, we'll probably get this fixed soon. So actually is the origin of SMPL unknown then? That's actually been a question that I've had for a while :) I.e. is applying the transformation to the vertices directly the only solution?

**Comment by kaufManu** [comment]
Fixed with 8ed00d33ccd12449fece9714f072ed7d632795ec.

**Comment by MarilynKeller** [comment]
Let me try to detail:

Let's define:
V: Transformed SMPL vertices
T(\beta): Template vertices, with beta applied
J_0 :  Root joint location for unposed T(\beta) (Note that this is subject specific)
R^(J_0): Root joint rotation matrix

SMPL applies the root joint at the template origin (so apply a -J_0 translation, apply the rotation and then do +J_0) eq. (1).
So in order to rotate the result, we do eq. (2), which is equivalent to eg. (3).

In a summary, if you want to do the rotation in load_from_amass, you will have to do eq. (4) and eq. (5). The only difference with what you had before is this J0. And indeed the offset I observed had the value (R^{y_{up}}(J_0) 

To get J_0 in load_from_amass, you'll have to compute a SMPL forward pass to get the location of J0 in Tpose (V, J SMPL(beta, pose=0) , which is a bit unconvenient because the class object is not initialized yet. I don't like that it is in the fk() either but having it in load_from_amass required a paragraph of explanation so up to  you :).

![image](https://user-images.githubusercontent.com/102662583/181822828-23ef5998-c1ad-4559-8d8d-82fb17021608.png)

**Comment by MarilynKeller** [comment]
And you can recover the location of the origin of SMPL at anytime by doing J, V = SMPL(beta, theta), smpl_origin = J[0] - trans.

**Comment by kaufManu** [comment]
Thank you for the detailed explanation! We've kept it in the `fk` function as you originally suggested to avoid a forward pass necessary to obtain the origin (because it's a bit easier to explain 😄).

The fix is included in the latest version (1.1, `pip install --upgrade aitviewer`).
